src/rt/monitor_.d

```diff
@@ -57,6 +57,11 @@ extern (C) void _d_monitordelete(Object h, bool det)
 }
 
 extern (C) void _d_monitorenter(Object h)
+in
+{
+    assert(h !is null, "Synchronized lock must not be null.");
```

Comment:
lock -> object

The argument of `_d_monitorenter` can be any class object. Synchronization primitives (like Mutex) are just a special case of objects, whose `__monitor` field indirectly points to themselves.